src/main/java/org/takes/facets/auth/social/PsLinkedin.java

```diff
@@ -49,40 +45,36 @@
  * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
  * @version $Id$
  * @since 0.11.3
+ * @checkstyle LineLength (500 lines)
  */
-@EqualsAndHashCode(of = { "app", "key" })
+@EqualsAndHashCode(of = { "luserjson" })
 public final class PsLinkedin implements Pass {
     /**
-     * App name.
+     * Social network member profile in json format.
      */
-    private final transient String app;
-
-    /**
-     * Key.
-     */
-    private final transient String key;
+    private final transient MemberProfileJson luserjson;
 
     /**
      * Ctor.
-     * @param lapp Linkedin app
-     * @param lkey Linkedin key
+     * @param userjson Member profile json
      */
-    public PsLinkedin(final String lapp, final String lkey) {
-        this.app = lapp;
-        this.key = lkey;
+    public PsLinkedin(final MemberProfileJson userjson) {
+        this.luserjson = userjson;
     }
 
     @Override
     public Iterator<Identity> enter(final Request request)
         throws IOException {
-        final Href href = new RqHref.Base(request).href();
-        // @checkstyle MultipleStringLiteralsCheck (1 line)
-        final Iterator<String> code = href.param("code").iterator();
-        if (!code.hasNext()) {
-            throw new IllegalArgumentException("code is not provided");
-        }
+        // @checkstyle LineLength (6 line)
         return Collections.singleton(
-            this.fetch(this.token(href.toString(), code.next()))
+            PsLinkedin.parse(this.luserjson
```

Comment:
@ikhvostenkov you shouldn't leave the parameter in the same line as `(` if `)` is not there.

So this line should be more like:
```
PsLinkedin.parse(
    this.luserjson.fetch(
        new Href(
            "...
```